tests/testapp/home/static/button_entity.js

```diff
@@ -0,0 +1,6 @@
+const ButtonDecorator = ({ entityKey, children }) => {
+  const attrs = {'data-tooltip': entityKey, className: 'RichEditor-button'};
+  return window.wagtaildraftail.createElement('span', attrs, children);
+};
```

Comment:
I think it's too brittle. To write even this small amount of JS/React without a compilation step, this means:

- Loading React on the page in a global variable.
- Writing React code without JSX.
- Writing JS code without module bundling.
- Writing JS code without transpiling.

In 2017 all of those things are bad practice IMHO. For example, this `ButtonDecorator` uses ES6 object destructuring and arrow functions so it won't work in IE11.